MovieApp/MovieApp/App/Features/MovieBrowser/Components/TableViewCategoryCell/CategoryCell.swift

```diff
@@ -52,42 +57,45 @@ class CategoryCell: UITableViewCell {
         categoryLabel.text = category.title
         genresView.populate(with: getGenres(category))
     }
-
-}
-
-extension CategoryCell: UICollectionViewDataSource {
     
-    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-        movies?.count ?? 0
+    private func makeDataSource() -> DataSource {
+        DataSource(
+            collectionView: moviesCollectionView,
+            cellProvider: { (collectionView, indexPath, movie) -> UICollectionViewCell? in
```

Comment:
You need to weakly capture `self` here as well.
The thing is that the closure captures `self` from their context, which is in this case `cellProvider` closure. For the `sink`s to be able to weakly capture `self` in their closures, they use strongly captured `self` from `cellProvider` closure.